pkgs/applications/editors/lite-xl/default.nix

```diff
@@ -0,0 +1,132 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, SDL2
+, lua52Packages
+, pkg-config
+, makeWrapper
+, openlibm
+, meson
+, pcre2
+, freetype
+, git
+, reproc
+, agg
+, ninja
+, curl
+, unzip
+, writeScriptBin
+}:
+let
+  build = writeScriptBin "build" ''
+    get_os_kind() {
+      if [[ "$OSTYPE" == "darwin"* ]]; then
+        echo "macos"
+      elif [[ "$OSTYPE" == "linux"* || "$OSTYPE" == "freebsd"* ]]; then
+        echo "linux"
+      else
+        echo "UNSUPPORTED-OS"
+      fi
+    }
+
+    main() {
+      local os=$(get_os_kind)
+      if [[ $os == "UNSUPPORTED-OS" ]]; then
+        echo "ERR: unknown OS type: \"$OSTYPE\""
+        exit 1
+      fi
+
+      local arch=$(uname -m)
+      local build_dir=build
+      local dest_dir=lite-xl
+      local debug
+      local force_fallback
+
+      [[ -d "$build_dir" ]] && rm -rf "$build_dir"
+
+      meson setup \
+        --prefix "$prefix" \
+        $force_fallback \
+        "$build_dir"
+
+      meson compile -C "$build_dir"
+
+      local data_dir="$(pwd)/$dest_dir/share/lite-xl"
+      local executable="$(pwd)/$dest_dir/bin/lite-xl"
+      local stripcmd="strip"
+
+      # Clean up previous package and directory
+      [[ -d "$dest_dir" ]] && rm -rf "$dest_dir"
+      [[ -f "$dest_dir.tar.gz" ]] && rm "$dest_dir".tar.gz
+      [[ -f "$dest_dir.zip" ]] && rm "$dest_dir".zip
+      mkdir -p "$data_dir"
+
+      # Copy third party modules
+      cp ${builtins.fetchurl {
+        url = https://github.com/rxi/lite-colors/archive/5b77292.zip;
+        sha256 = "09s51kfqvajsnc5ywyvrwn68imdfb8vz33zvmv4sqwpw387xnfiz";
+      }} "$build_dir/rxi-lite-colors.zip"
+
+      mkdir -p "$build_dir/third/data/colors"
+      unzip "$build_dir/rxi-lite-colors.zip" -d "$build_dir"
+      mv "$build_dir/lite-colors-master/colors" "$build_dir/third/data"
+      rm -rf "$build_dir/lite-colors-master"
+
+      for module_name in colors; do
+        cp -r "$build_dir/third/data/$module_name" "$data_dir"
+      done
+
+      DESTDIR="$(pwd)/$dest_dir" meson install -C "$build_dir"
+
+      $stripcmd "$executable"
+
+      #tar czvf "$dest_dir-$os-$arch".tar.gz "$dest_dir"
+    }
+
+    main "$@"
+  '';
+in
+stdenv.mkDerivation rec {
+  pname = "lite-xl";
+  version = "2.0.1";
+
+  src = fetchFromGitHub {
+    owner = "lite-xl";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-+RbmT6H/5Ldhv3qOClxMjCSGMudbkGtkjo2SpGqExao=";
+  };
+
+  nativeBuildInputs = [ makeWrapper pkg-config meson ninja curl unzip ];
+
+  buildInputs = [ SDL2 lua52Packages.lua openlibm pcre2 freetype git agg reproc build ];
+
+  patches = [
+    # this patch makes meson fallback to local dependency instead of fetching it.
+    ./meson-fallback.patch
+    # this patch replaces unpack() with table.unpack(). this should be removed when the next release is proposed (2.0.2). Check PR #435.
+    ./table-unpack.patch
+  ];
+
+  # Nix automatically tries to run meson
+  configurePhase = " ";
```

Comment:
Maybe `dontUseMesonConfigure` (not familiar with this package, are you sure we have to do this?)

Reply:
never heard of that option at all, should've checked the docs!